### PR TITLE
CI improvements

### PR DESCRIPTION
**Pull request by jjanowsk**
### Description
Few improvements to the CI:
1. Gitlab pipeline is run automatically for PRs where feature branch is residing inside NordSecurity repository. This way there is no need to add `run tests` label in order to trigger gitlab pipelines for internal PRs.
2. build, linters, tests and fuzzing were put into one `CI` workflow in order to use `check-all-green` and `skip-duplicate-actions` actions more easily
3. `skip-duplicate-actions` is used to skip actions if multiple of them are running for the same sha. We have this situation because we run our actions for both `push` and `pull_request` events. Alternatively we could also not run them for push (leave only push to master) but the advantage of keeping both of them is that you can push your feature branch and check the actions without needing to open a PR. I was sometimes using it on Gitlab and I think we should leave this behavior and only think about dropping it if it turns out that this `skip-duplicate-actions` does not work correctly.
4. `check-all-green` is used to collect status checks from multiple jobs in order to easily add it into the settings under: `Require status checks to pass before merging`. Without it we would need to specify ~30 of them there and modify every time we change our CI.
5. Top level permission added to `dod.yml`

### :ballot_box_with_check: Definition of Done checklist
- [x] Commit history is clean
- [x] README.md is updated
- [x] changelog.md is updated

### CODE REVIEW

**Comment by gytsto** [comment]
`check-all-green` doesn't seem to check `Gitlab` job status or am I missing something?

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  skipper:
+    runs-on: ubuntu-latest
+    permissions:
+      actions: write
+      contents: read
+    outputs:
+      should_skip: ${{ steps.skip_check.outputs.should_skip }}
+    steps:
+      - id: skip_check
+        uses: fkirc/skip-duplicate-actions@v5
```
Comment (lukjok) [inline .github/workflows/ci.yml:15]: Change version to the commit SHA

**Review by gytsto — APPROVED** [review]
+1

File: .github/workflows/ci.yml (hunk 1 above)
Comment (jjanowsk) [reply .github/workflows/ci.yml]: Done

**Review by tomaszklak — APPROVED** [review]
+1, lgtm

**Pull request merged**